### PR TITLE
Improvements around setting up development environment and running nodes

### PR DESCRIPTION
**Pull request by fabrice102**
* add hardware requirements and details for fast catchup, including requirement for non-archival node and troubleshooting in case of failure
* Slight refactoring of the devenv setup
  * Split sandbox on a new page
  * Recommend installing the Algorand software even when not running a node
  * Link to the list of API services

minor:
* add missing blank lines in AVM document

### CODE REVIEW

**Comment by nullun** [comment]
Looks good. I'm a fan of introducing the Sandbox sooner as it's pretty key these days. Also encouraging to install locally is a good idea! I'll wait for another pair of eyes before merging.

**Review by ryanRfox — APPROVED** [review]
Appreciate these updates

**Pull request merged**
